src/components/AuthenticationPage.js

```diff
@@ -2,29 +2,50 @@ import Greeting from "./Greeting";
 import Header from "./Header";
 import AuthenticationForm from "./AuthenticationForm";
 
+const signup = {
+  title: "Sign Up",
+  linkMessage: "Have an account?",
+  link: "#",
+  placeholders: ["Username", "Email", "Password"],
```

Comment:
It doesn't make sense to separate placeholders name and input type like this in different array.  Combine them all into an array of objects. It will make some of your downstream logic more readable as well (such as `placeholder.map`)

Reply:
This was very helpful. Changing the data structure made the code much cleaner and easier to read. Thanks